plugins/scanners/scanoss/src/main/kotlin/ScanOssResultParser.kt

```diff
@@ -144,9 +144,9 @@ private fun getSnippets(details: ScanFileDetails): Set<Snippet> {
     val url = requireNotNull(details.url)
     val purls = requireNotNull(details.purls)
 
-    val licenses = details.licenseDetails.orEmpty().mapTo(mutableSetOf()) { license ->
-        SpdxExpression.parse(license.name)
-    }
+    val license = details.licenseDetails.orEmpty()
+        .map { license -> SpdxExpression.parse(license.name) }
+        .toExpression()?.sorted() ?: SpdxLicenseIdExpression(SpdxConstants.NOASSERTION)
```

Comment:
This seems to be an optimization that's unrelated to the strict topic of this commit. I'd extract it to a preceding commit saying

```
chore(scanoss): Determine the snippet license only once

The `licenses` variable is constant in the loop, so extract it to determine
the resulting `license` only once.
```

Reply:
To clarify, I've checked that the added tests also pass without this change. While it's good to have more test coverage, I'm unsure whether that's what you intended to do.

Reply:
> This seems to be an optimization that's unrelated to the strict topic of this commit. I'd extract it to a preceding commit saying

I'll reword it as suggested.

Reply:
> To clarify, I've checked that the added tests also pass without this change. While it's good to have more test coverage, I'm unsure whether that's what you intended to do.

Tests were included to verify edge cases with duplicate licenses from differences sources and empty license arrays. 

I'll move them to a separate commit preceding the changes with this message:

```
test(scanoss): Add tests for license handling edge cases

Add test cases to verify behavior when combining the same license from 
different sources and handling empty license arrays with NOASSERTION value.
```